scripts/utils.py

```diff
@@ -0,0 +1,169 @@
+import yaml
+
+from google.cloud import bigquery
+
+
+def parse_arguments(argv):
+  """Parses command line arguments common across scripts."""
+  parser = argparse.ArgumentParser(description='Configuration options..')
+  parser.add_argument(
+      '--config_path',
+      required=True,
+      type=str,
+      help='YAML file with project configuration.')
+  args, _ = parser.parse_known_args(args=argv[1:])
+  return args
+
+
+def merge_dicts(a: Dict[str, Any], b: Dict[str, Any]) -> Dict[str, Any]:
+  """Return the union of two dicts, raise an error if keys are non-unique."""
+  # Resolve None types.
+  if a is None:
+    if b is None:
+      return {}
+    return b
+  if b is None:
+    return a
+
+  duplicate_keys = set(a.keys()).intersection(set(b.keys()))
+  if duplicate_keys:
+    duplicate_keys_string = ", ".join(duplicate_keys)
+    raise KeyError("The following keys are non-unique: {}".format(
+        duplicate_keys_string))
+
+  return {**a, **b}
+
+
+def read_config(config_path: str,
+                subconfig_keys: List[str] = None) -> Dict[str, Any]:
+  """Reads from a YAML config file to generate a dict of parameters.
+
+  Args:
+    config_path: Path to YAML config file.
+    subconfig_keys: List of keys for subconfigs (second level) to extract.
+        None will return the top level.
+
+  Returns:
+    Dict of parameters, which may have a nested structure if
+        subconfig_keys is None.
+
+  Raises:
+    KeyError if any keys are repeated across the subconfigs when merging.
+    KeyError if any of the subconfig keys are not in the config.
+  """
+
+  with open(config_path, 'r') as f:
+    config = yaml.safe_load(f)
+
+  # Return full nested config if no subconfigs are specified.
+  if not subconfig_keys:
+    return config
+
+  # Merge all specified subconfigs into a single config.
+  merged_config = {}
+  for subconfig_key in subconfig_keys:
+    subconfig = config.get(subconfig_key)
+    if not subconfig:
+      raise KeyError('Key "{}" does not exist.'.format(subconfig_key))
+    merged_config = merge_dicts(merged_config, subconfig)
+
+  return merged_config
+
+
+def create_dataset(destination_project: str, destination_dataset: str) -> str:
+  """Creates a Bigquery dataset.
+
+  Args:
+    destination_project: GCS project to create the dataset in.
+    destination_dataset: Bigquery dataset to create.
+
+  Uses application default credentials.
+  Raises google.api_core.exceptions.Conflict if the dataset already exists.
+  """
+
+  # Create dataset in Bigquery.
+  client = bigquery.Client()
+  dataset_id = '{}.{}'.format(destination_project,
+                              destination_dataset)
+  dataset_ref = bigquery.Dataset(dataset_id)
+  result = client.create_dataset(dataset_ref)
+
+  return result
+
+
+def create_table(
+    query_path: str,
+    query_params: Dict[str, Any],
+    destination_project: str,
+    destination_dataset: str,
+    destination_table: str,
+    partition_field: str,
+) -> str:
```

Comment:
I'm surprised this passed the linter. Should be on the previous line (unless it won't let you).